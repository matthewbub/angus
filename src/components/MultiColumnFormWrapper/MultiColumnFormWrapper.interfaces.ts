export interface MultiColumnFormWrapperProps {
  children: React.ReactNode;
  title: string;
  description?: string;
};