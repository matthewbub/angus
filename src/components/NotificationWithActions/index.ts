export { default as NotificationWithActions } from './NotificationWithActions';