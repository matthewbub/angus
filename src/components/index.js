export { AuthWrapper } from './AuthWrapper';
export { Button } from './Button';
export { ConfigurableForm } from './ConfigurableForm';
export { ConsoleLayout } from './ConsoleLayout';
export { FieldError } from './FieldError';
export { FieldWrapper } from './FieldWrapper';
export { FullNavigation } from './FullNavigation';
export { Input } from './Input';
export { MultiColumnFormWrapper } from './MultiColumnFormWrapper';
export { NotificationWithActions } from './NotificationWithActions';
export { Select } from './Select';
export { SideNavigation } from './SideNavigation';
export { SlideOver } from './SlideOver';
export { Stats } from './Stats';
export { TextArea } from './TextArea';
export { Modal } from './Modal';
