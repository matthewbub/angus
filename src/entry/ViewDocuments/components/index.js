export { default as NewDocumentForm } from './NewDocumentForm';
export { default as TableDisplay } from './TableDisplay';