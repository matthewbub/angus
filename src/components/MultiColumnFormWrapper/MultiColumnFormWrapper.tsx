import React, { FC } from 'react';
import { MultiColumnFormWrapperProps } from './MultiColumnFormWrapper.interfaces';

const MultiColumnFormWrapper: FC<MultiColumnFormWrapperProps> = ({ title, description, children }) => {
  return (
    <div className="grid max-w-7xl grid-cols-1 gap-x-8 gap-y-10 px-4 py-16 sm:px-6 md:grid-cols-3 lg:px-8">
      <div>
        {title && title.length > 0 && (
          <h2 className="text-base font-semibold leading-7 txt1" >
            {title}
          </h2>
        )}
        {description && description.length > 0 && (
          <p className="mt-1 text-sm leading-6 text-gray-400">
            {description}
          </p>
        )}
      </div>

      {children}
    </div>
  )
}

export default MultiColumnFormWrapper;