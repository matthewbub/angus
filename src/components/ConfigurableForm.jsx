import { useState } from 'react'
import { useForm } from 'react-hook-form';
import { get } from 'lodash';

import { SlideOver } from '@/src/components/SlideOver';
import { Modal } from '@/src/components';
import TextArea from '@/src/components/TextArea';
import { Button, Select, Input } from '@/src/components';

const SubmitButton = ({ children }) => (
  <div className='col-span-4'>
    <Button className='text-center mt-12'>
      {children || 'Submit'}
    </Button>
  </div>
)

export const ConfigurableForm = ({
  form,
  open,
  setOpen,
  slideOverTitle,
  submitForm = (formFields) => { return formFields; },
}) => {
  const { register, handleSubmit } = useForm();
  const formFields = get(form, 'formFields', []);

  return (
    <SlideOver
      open={open}
      setOpen={setOpen}
      title={slideOverTitle}
    >
      <form className='grid grid-cols-12 gap-5' onSubmit={handleSubmit(submitForm)}>
        {formFields.map((field, index) => {
          let Field = null;
          switch (field.field) {
            case 'Divider':
              return (
                <div className='col-span-12 mt-2 mb-12'>
                  <div className='border-b border-white/20 h-2' />
                </div>
              );
            case 'Input':
              Field = Input;
              break;
            case 'TextArea':
              Field = TextArea;
              break;
            case 'Select':
              Field = Select;
              break;
            case 'SubmitButton':
              return (
                <SubmitButton>
                  {field.label}
                </SubmitButton>
              );
            default:
              Field = Input;
              break;
          }

          return (
            <Field
              key={index}
              label={field.label}
              name={field.name}
              type={field.type}
              error={field.error || null}
              register={register}
              className={field.className}
              options={field.options}
              registerOptions={field.validate}
              defaultValue={field.defaultValue}
            />
          )
        })}
      </form>
    </SlideOver>
  )
}