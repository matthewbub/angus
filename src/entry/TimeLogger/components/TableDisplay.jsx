import React from 'react';
import { Stats } from '../../../components';

const TableDisplay = ({ data = [] }) => {
  return (
    <>
      <Stats
        data={[
          {
            name: 'Total Documents',
            value: '-'
          },
          {
            name: 'Projects',
            value: '-',
          }
        ]}
        loading={false}
      />
      <table className="w-full whitespace-nowrap text-left">
        <colgroup>
          <col className="w-full sm:w-5/12" />
          <col className="lg:w-4/12" />
          <col className="lg:w-2/12" />
          <col className="lg:w-1/12" />
        </colgroup>
        <thead className="border-b border-neutral-900 text-sm leading-6 txt1 bg-neutral-900 sticky top-[63px]">
          <tr>
            <th scope="col" className="py-2 pl-4 pr-8 font-semibold sm:pl-6 lg:pl-8">
              Project
            </th>
            <th scope="col" className="hidden py-2 pl-0 pr-8 font-semibold sm:table-cell">
              Owner
            </th>
            <th scope="col" className="hidden py-2 pl-0 pr-4 text-right font-semibold sm:table-cell sm:pr-8 sm:text-left lg:pr-20">
              Hours Spent
            </th>
            <th scope="col" className="hidden py-2 pl-0 pr-8 font-semibold md:table-cell lg:pr-20">
              <span className="sr-only">Edit</span>
            </th>
          </tr>
        </thead>
        <tbody className="divide-y divide-white/20">
          {data && data.length > 0 && data.map((document) => (
            <tr key={document.id}>
              <td className="py-4 pl-4 pr-8 sm:pl-6 lg:pl-8">
                <span
                  className='truncate block w-full max-w-screen-sm'
                  style={{ maxWidth: width - 40 + 'px' }}
                >{document.document_title || 'Untitled Document'}</span>
              </td>
              <td className="hidden py-4 pl-0 pr-4 sm:table-cell sm:pr-8">
                {document.document_owner}
              </td>
              <td className="hidden py-4 pl-0 pr-4 sm:table-cell text-sm leading-6 sm:pr-8 lg:pr-20">
                {document.created_at}
              </td>
              <td className="hidden py-4 pl-0 pr-8 text-sm leading-6 text-gray-400 md:table-cell lg:pr-20">
                <a href={"/documents/" + document.id} className="text-teal-400 hover:text-teal-600">
                  Edit<span className="sr-only">, {document.document_title}</span>
                </a>
              </td>
            </tr>
          ))}
        </tbody>
      </table>

    </>
  )
}

export default TableDisplay;