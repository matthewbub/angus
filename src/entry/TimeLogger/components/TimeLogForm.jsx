import { Input, Button, TextArea, Select } from '../../../components';
import { Fragment } from "react"
import { useForm } from 'react-hook-form';

const TimeLogForm = () => {
  const { register } = useForm();
  return (
    <Fragment>
      <div className=''>
        <h1 className='text-base font-semibold leading-7 txt1'>Quick Log</h1>
        <p className='mt-1 max-w-2xl text-sm leading-6 text-gray-400'>Add time spent working on a task.</p>
      </div>
      <div className='max-w-md grid gird-cols-12'>
        <Input
          label='Title'
          register={register}
          name="title"
          className='col-span-9'
        />
        <Input
          label='Date'
          register={register}
          name="date"
          type="datetime-local"
          className='col-span-9'
        />
        <Input
          label='Time Spent'
          register={register}
          name="time-spent"
          className='col-span-9'
          placeholder='HH:MM'
        />
        <TextArea
          label='Description'
          register={register}
          name="description"
          className='col-span-12'
          placeholder='Describe what you did.'
        />
        <div className='col-span-12 pt-10'>
          <Button>
            Log Time
          </Button>
        </div>
      </div>
    </Fragment>
  )
}

export default TimeLogForm;