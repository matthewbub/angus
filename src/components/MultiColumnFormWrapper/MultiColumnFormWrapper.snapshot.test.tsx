import { render } from '@testing-library/react'
import { MultiColumnFormWrapper } from '@/components/MultiColumnFormWrapper'

const HelloWorld = () => <>{'I\'m a make-believe form!'}</>
const MockMultiColumnFormWrapper = () => (
  <MultiColumnFormWrapper
    title="Hello, World!"
    description="I'm a description!"
  >
    <HelloWorld />
  </MultiColumnFormWrapper>
);

it('renders MultiColumnFormWrapper unchanged', () => {
  const { container } = render(<MockMultiColumnFormWrapper />)
  expect(container).toMatchSnapshot()
});