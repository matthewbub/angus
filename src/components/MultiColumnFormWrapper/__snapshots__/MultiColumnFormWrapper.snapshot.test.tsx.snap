// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MultiColumnFormWrapper unchanged 1`] = `
<div>
  <div
    class="grid max-w-7xl grid-cols-1 gap-x-8 gap-y-10 px-4 py-16 sm:px-6 md:grid-cols-3 lg:px-8"
  >
    <div>
      <h2
        class="text-base font-semibold leading-7 txt1"
      >
        Hello, World!
      </h2>
      <p
        class="mt-1 text-sm leading-6 text-gray-400"
      >
        I'm a description!
      </p>
    </div>
    I'm a make-believe form!
  </div>
</div>
`;
